docs/development/pigs/pig-008.rst

```diff
@@ -0,0 +1,449 @@
+on the parallel evaluation, because Python objects can't easily be shared between
+multiple processes. We are aware of this issue, but propose to solve it later,
+because it does not affect the proposed API for datasets.
+
+
+Dataset serialization
+---------------------
+
+For convenience all dataset classes should support serialization, implemented
+via `.read()` and `.write()` methods. As the dataset as to orchestrate the
+serialization of mutiple objects, such as model, maps, flux-points etc. the best
+option is likely to introduce the serialization with a YAML based index file:
+
+.. code ::
+
+    dataset = MapDataset.read("dataset.yaml") # lazy loading?
+    dataset.write("dataset.yaml")
+
+Where the index file points to the various files needed for initialization of the
+dataset. Here is an example:
+
+.. code::
+
+    dataset:
+        type: map-dataset
+        counts: "obs-123/counts.fits"
+        exposure: "obs-123/exposure.fits"
+        edisp: "obs-123/edisp.fits"
+        psf: "obs-123/psf.fits"
+        model: "model.yaml"
+        background-model: "obs-123/background.fits"
+
+Addtionally one could introduce a single FITS file serializiaton for quickly writing /
+reading datasets to disk.
+
+The `Datasets` object could be serialized equivalently as a list of datasets:
+
+.. code::
+
+    - dataset:
+        type: spectrum-dataset
+        maps:
+            counts: "obs-123/counts.fits"
+            exposure: "obs-123/exposure.fits"
+            edisp: "obs-123/edisp.fits"
+        model: "model.yaml"
+        background-model: "obs-123/background.fits"
+        likelihood: "wstat"
+
+    - dataset:
+        type: spectrum-dataset
+        maps:
+            counts: "obs-124/counts.fits"
+            exposure: "obs-124/exposure.fits"
+            edisp: "obs-124/edisp.fits"
+        model: "model.yaml"
+        background-model: "obs-124/background.fits"
+        likelihood: "wstat"
+
+    - dataset:
+        type: flux-point-dataset
+        flux-points : "fermipy-flux-points.fits"
+        model: "spectral-model.yaml"
+        likelihood: "template"
+
+
+Lazy loading of Datasets
+------------------------
+For the use-case of inspecting individual datasets (e.g. MapDataset per observation)
+it could be advantegeous to implement a lazy-loading or generator interface for
+`Datasets.read()`. Such that the individual datasets are only read from disk on
+requests and are not loaded in memory when calling `.read()`. We propose to handle
+this with a generic lazy-loading mechanism for `Map` objects, where the data is loaded
+on access of the `.data` attribute.
+
+
+Simulation of `MapDataset`
+--------------------------
+The `MapDataset` class will be the central class for map based analyses. In many
+cases it's useful to simulate counts from a predicted number of counts model. Two
+approaches are possible: simulating a counts map by calling `np.random.poisson()` on
```

Comment:
This paragraph is a bit hard to read. Suggest to just describe the poisson fluctuation as "binned simulation" first, and then after the code example mention the second option of sampling events.
Wouldn't the API probably be just `dataset.simulate_events`?